tests/test_context.py

```diff
@@ -1,4 +1,26 @@
-from nnbench.context import CPUInfo, GitEnvironmentInfo, PythonInfo
+import pytest
+
+from nnbench.context import Context, CPUInfo, GitEnvironmentInfo, PythonInfo
+
+# Sample context dicts for testing
+nested_dict = {"a": 1, "b": {"c": 2, "d": {"e": 3}}}
+
+flat_dict = {"a": 1, "b.c": 2, "b.d.e": 3}
+
+
+@pytest.fixture
+def context_instance():
+    return Context()
+
+
+@pytest.fixture
+def flat_context_instance():
+    return Context(flat_dict)
+
+
+@pytest.fixture
+def nested_context_instance():
+    return Context(nested_dict)
```

Comment:
I think you can provision them as necessary, if anything, the data above this should be a fixture.